[pdal] Build error

Package: pdal[core,draco,e57,hdf,i3s,pgpointcloud,supported-plugins]:x64-linux -> 2.4.3

**Host Environment**

- Host: x64-linux
- Compiler: GNU 8.5.0
-    vcpkg-tool version: 2022-09-20-522aa94e9d261c7d7b2f079bf2591ca62df5c714
    vcpkg-scripts version: e2667a41f 2022-10-06 (14 hours ago)

**To Reproduce**

`vcpkg install pdal:x64-linux`

**Failure logs**

```
-- Note: pdal only supports dynamic library linkage. Building dynamic library.
-- Using cached PDAL-PDAL-2.4.3.tar.gz.
-- Cleaning sources at /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/src/2.4.3-960ddf7f39.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/ec2-user/test_vcpkg/vcpkg/downloads/PDAL-PDAL-2.4.3.tar.gz
-- Applying patch fix-dependency.patch
-- Applying patch fix-unix-compiler-options.patch
-- Applying patch fix-find-library-suffix.patch
-- Applying patch no-pkgconfig-requires.patch
-- Applying patch no-rpath.patch
-- Using source at /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/src/2.4.3-960ddf7f39.clean
-- Installing: /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/src/2.4.3-960ddf7f39.clean/vendor/nanoflann/nanoflann.hpp
-- Installing: /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/src/2.4.3-960ddf7f39.clean/vendor/nlohmann/nlohmann/json.hpp
-- Installing: /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/src/2.4.3-960ddf7f39.clean/pdal/json_fwd.hpp
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:129 (message):
    Command failed: /home/ec2-user/test_vcpkg/vcpkg/downloads/tools/cmake-3.24.0-linux/cmake-3.24.0-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j9
    Working Directory: /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/x64-linux-dbg
    See logs for more information:
      /home/ec2-user/test_vcpkg/vcpkg/buildtrees/pdal/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/pdal/portfile.cmake:68 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)

~~~
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::path::parent_path() const'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::directory_iterator::operator*() const'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::remove(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `typeinfo for std::filesystem::__cxx11::filesystem_error'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::path::has_root_directory() const'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::file_size(std::filesystem::__cxx11::path const&, std::error_code&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::current_path[abi:cxx11]()'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::status(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::create_directories(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::remove_all(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::directory_iterator::operator++()'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::rename(std::filesystem::__cxx11::path const&, std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::create_directory(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::directory_iterator::directory_iterator(std::filesystem::__cxx11::path const&, std::filesystem::directory_options, std::error_code*)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::absolute(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::weakly_canonical(std::filesystem::__cxx11::path const&)'
lib/libpdal_util.so.14.3.0: undefined reference to `std::filesystem::__cxx11::path::has_filename() const'
~~~
https://github.com/PDAL/PDAL/issues/3826

--- a/ports/pdal/fix-gcc8-compatibility.patch
+++ b/ports/pdal/fix-gcc8-compatibility.patch
@@ -1,0 +1,15 @@
+diff --git a/pdal/util/CMakeLists.txt b/pdal/util/CMakeLists.txt
+index 5b48ba8..e031e2b 100644
+--- a/pdal/util/CMakeLists.txt
++++ b/pdal/util/CMakeLists.txt
+@@ -46,3 +46,10 @@ set_target_properties(${PDAL_UTIL_LIB_NAME} PROPERTIES
+     CLEAN_DIRECT_OUTPUT 1)
+ 
+ set_property(GLOBAL PROPERTY _UTIL_INCLUDED TRUE)
++
++# Compatibility for GCC8: std::filesystem requires linking with libstdc++fs
++target_link_libraries(
++  ${PDAL_UTIL_LIB_NAME}
++  PRIVATE
++  "$<$<AND:$<CXX_COMPILER_ID:GNU>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,9.0>>:-lstdc++fs>"
++)

--- a/ports/pdal/portfile.cmake
+++ b/ports/pdal/portfile.cmake
@@ -12,6 +12,7 @@ vcpkg_from_github(
         fix-find-library-suffix.patch
         no-pkgconfig-requires.patch
         no-rpath.patch
+        fix-gcc8-compatibility.patch #Upstream PR: https://github.com/PDAL/PDAL/pull/3864
 )
 
 # Prefer pristine CMake find modules + wrappers and config files from vcpkg.

--- a/ports/pdal/vcpkg.json
+++ b/ports/pdal/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "pdal",
   "version": "2.4.3",
+  "port-version": 1,
   "description": "PDAL - Point Data Abstraction Library is a library for manipulating point cloud data.",
   "homepage": "https://pdal.io/",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5618,7 +5618,7 @@
     },
     "pdal": {
       "baseline": "2.4.3",
-      "port-version": 0
+      "port-version": 1
     },
     "pdal-c": {
       "baseline": "2.1",

--- a/versions/p-/pdal.json
+++ b/versions/p-/pdal.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "be0fc52801df7c7409e934914c7cca650f507f8e",
+      "version": "2.4.3",
+      "port-version": 1
+    },
+    {
       "git-tree": "8b72a995f43fed0287dca94b7f45bc1855b8fa43",
       "version": "2.4.3",
       "port-version": 0

